sdk/python/feast/infra/registry/caching_registry.py

```diff
@@ -37,6 +37,7 @@ def __init__(self, project: str, cache_ttl_seconds: int, cache_mode: str):
         )
         self.cached_registry_proto = self.proto()
         self.cached_registry_proto_created = _utc_now()
+        self._refreshing = False  # Prevents reentrant calls to refresh
```

Comment:
can we use `_refresh_lock` status instead of  having another property?

Reply:
Done!